### ISSUE TITLE
Python 3.7 exception (PEP 479)

### ISSUE THREAD

**Issue by RazerM** [ashafer01/laurelin#15]
Generators should `return` instead of `raise StopIteration`. This is backwards compatible.

https://www.python.org/dev/peps/pep-0479/#writing-backwards-and-forwards-compatible-code

Exception:
```
Traceback (most recent call last):
  File "/Users/.../python3.7/site-packages/laurelin/ldap/base.py", line 1580, in __iter__
    mid, entry, res_ctrls = unpack('searchResEntry', msg)
  File "/Users/.../python3.7/site-packages/laurelin/ldap/protoutils.py", line 44, in unpack
    raise UnexpectedResponseType('Got {0} but expected {1}'.format(got_op, op))
laurelin.ldap.exceptions.UnexpectedResponseType: Got searchResDone but expected searchResEntry

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/.../python3.7/site-packages/laurelin/ldap/base.py", line 1603, in __iter__
    raise StopIteration()
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/frazer/projects/work/sdsso/wip.py", line 6, in <module>
    ldap = LDAP('ldap://localhost:9389')
  File "/Users/.../python3.7/site-packages/laurelin/ldap/base.py", line 339, in __init__
    self.refresh_root_dse()
  File "/Users/.../python3.7/site-packages/laurelin/ldap/base.py", line 364, in refresh_root_dse
    self.root_dse = self.get('', ['*', '+'])
  File "/Users/.../python3.7/site-packages/laurelin/ldap/base.py", line 609, in get
    results = list(self.search(dn, Scope.BASE, attrs=attrs, limit=2, **kwds))
RuntimeError: generator raised StopIteration
```

**Comment by ashafer01** [comment]
Thanks for the report, PEP link was greatly appreciated. This has been fixed and 1.5.3 has been pushed to PyPI.

**Comment by ashafer01** [comment]
Side note: 3.7 is still [broken on travis CI](https://github.com/travis-ci/travis-ci/issues/9069) so I am currently only able to test against my local 3.7.
